lib/spack/spack/test/install.py

```diff
@@ -139,6 +141,24 @@ def test_installed_dependency_request_conflicts(
         dependent.concretize()
 
 
+def test_install_dependency_symlinks(
+        install_mockery, mock_fetch, mutable_mock_packages):
+    dependent = Spec('dependent-install')
+    dependent.concretize()
+    pkg = dependent.package
+    pkg.do_install()
+
+    # Ensure dependency directory appears only after the symlinks call
+    dependency_name = 'dependency-install'
+    assert dependency_name not in os.listdir(pkg.prefix)
+
+    install_dependency_symlinks(pkg, dependent, pkg.prefix)
+    assert dependency_name in os.listdir(pkg.prefix)
```

Comment:
(minor) I think you can remove this check because the next assert is more specific and I think provides nearly as much info as the two of them together.

Reply:
Fair enough.

Reply:
Removed.